File: catalog/admin/item/pub.php
<?php
defined('AUTH') or die('Restricted access');

include_once $_SERVER['DOCUMENT_ROOT'].'/components/catalog/admin/classes/AdminCatalogItem.php';
$ITEM = new AdminCatalogItem();

$arr['id'] = intval($SITE->d[5]);  // item id
$arr['pub'] = $SITE->d[4] == 'pub' ? 1 : 0;  // pub / unpub

$section_id = $ITEM->getItem($arr['id'])['section_id'];

$ITEM->pub_unpub($arr);

header('location: /admin/com/catalog/section/'.$section_id);

File: dan/system/catalog/item/pub.py
import sys
sys.path.append('system/catalog/classes')
from system.catalog.classes.Item import Item


def pub(SITE):
    print('FUNCTION -> system-> calalog -> item -> pub/unpub')
    type = SITE.p[2]
    id = SITE.p[3]

    ITEM = Item(SITE)
    section_id = ITEM.pub(type, id)

    if 'cancel' in SITE.post:
        return {'redirect': '/system/catalog/section/' + str(section_id)}
    return {'redirect': '/system/catalog/section/' + str(section_id)}